doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -0,0 +1,136 @@
+      
+    Fundamentally it means - have a deterministic approach to bring-up the interface.
+      
+    Also, this problem is mentioned ‘as outside-the-scope’ of ‘CMIS Application Initialization’ high-level design document
+      **(https://github.com/ds952811/SONiC/blob/0e4516d7bf707a36127438c7f2fa9cc2b504298e/doc/sfp-cmis/cmis-init.md#outside-the-scope)**
+
+2.  During administrative interface disable “config interface shutdown Ethernet”, only the ASIC(NPU) Tx is disabled and not the optics laser for 100G/400G. 
+      This will lead to power wastage and un-necessary fan power consumption to keep the module temperature in operating range 
+
+# Background
+
+  Per the ‘QSFPDD spec’,  ‘validation, diagnostics’ done by HW team' and 'agreement with vendors', 
+  need to follow following bring-up seq to enable port/interface with 400G optics in LC/chassis:
+
+    a) Enable port on NPU (bring-up port, serdes on the NPU ; enable signals) : syncd
+    b) Enable port on PHY (bring-up port, serdes on the PHY ; enable signals) : gbsyncd
+       - Wait for signal to stabilize on PHY   
+    c) Enable optical module (turn laser on/ enable tx) : xcvrd or platform bootstrap/infra
+
+  In boards not having PHY, #b) not needed but #a) and #c) sequence to be followed.
+  
+  ## Clause from QSFP-DD (CMIS4.0 spec)
+  
+     Excerpt from CMIS4.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![61f5b485-cf3b-4ca8-beac-9102b6feabfe](https://user-images.githubusercontent.com/69485234/147173702-f124fc9d-ef27-4816-b1a1-b4a44a5833a7.PNG)
+
+
+  ## Clause from QSFP-DD (CMIS5.0 spec)
+  
+     Excerpt from CMIS5.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![96a35dc5-618f-418c-9593-5639a90f1b28](https://user-images.githubusercontent.com/69485234/147173164-5ad0123c-479a-4774-b3ee-12a81fdd7d7e.PNG)
+     
+
+# Objective
+
+Have a determistic approach for Interface link bring-up sequence i.e. below sequence to be followed:
+  1. Initialize and enable NPU Tx and Rx path
+  2. For system with 'External' PHY: Initialize and enable PHY Tx and Rx on both line and host sides; ensure host side link is up 
+  3. Then only perform optics data path initialization/activation/Tx enable (for 400G) and Tx enable (for 100G) 
+
+# Proposal
+
+Recommend following this high-level work-flow sequence to accomplish the Objective:
+- xcvrd to subscribe to a new field “host_tx_ready” in port table state-DB
+- Orchagent will set the “host_tx_ready” to true/false based on the SET_ADMIN_STATE attribute return status to syncd/gbsyncd. (As part of SET_ADMIN_STATE attribute enable, the NPU Tx is enabled)
+- xcvrd process the “host_tx_ready” value change event and do optics datapath init / de-init using CMIS API
+- Recommendation is to follow this proposal for all the known interface speeds - 400G/100G/40G/25G/10G
+  - 400G - as mentioned above the CMIS spec to be followed
```

Comment:
This should be based on module type, not interface speed.

Reply:
what about 400g DAC cable? does xcvrd need to process host_tx_ready?

Reply:
@mikeberesford 
> This should be based on module type, not interface speed.
Removed 'speed'.
Reason - the sync flow is independent of transceiver type present in the port. 
host_tx_ready would be set based on host (NPU, PHY) Tx status.
xcvrd on receiving this host_tx_ready notification will ignore or program the transceiver (DP init/ laser enable) based on transceiver presence/type.

@lguohan 
> what about 400g DAC cable? does xcvrd need to process host_tx_ready?
For 400G DAC cable, xcvrd would ignore host_tx_ready notification

Reply:
The speed should be totally left out of this statement. This should be based on the module type - ie for all CMIS modules (regardless of configured speed) per the CMIS spec.

Reply:
Just to clarify the entire synchronization applies to any Interface regardless of transceiver presence. 
I will change the sub-heading 400G and 100G/40G/25G/10G to CMIS complaint optical modules and SFF complaint optical modules respectively.